### PR TITLE
Discuss the impact of resolver selection on security

### PR DESCRIPTION
**Pull request by bemasc**

### CODE REVIEW

File: draft-ietf-tls-svcb-ech.md [hunk 1]
```diff
@@ -90,12 +90,14 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
...
 
 In an idealized deployment, ECH protects the SNI with an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server. Accordingly, an attacker who can enumerate this set can always guess the encrypted SNI with probability 1/K, where K is the number of domains in the set. In practice, this probability may be increased via traffic analysis, popularity weighting, and other mechanisms.
 
+The name of the destination domain is protected from disclosure by ECH, but it is still revealed by the DNS queries used to reach the server.  To achieve the security goals from {{Section 10.1 of !ECH}}, clients for this specification SHOULD select a DNS resolver that they trust to preserve the confidentiality of their queries and return authentic answers, and use authenticated encryption for DNS queries and responses.
```
Comment (richsalz) [inline draft-ietf-tls-svcb-ech.md]: looks good to me.

Comment (MikeBishop) [reply draft-ietf-tls-svcb-ech.md]: Content is good, but the comma makes the second sentence parse oddly.  What about...

```suggestion
The name of the destination domain is protected from disclosure by ECH, but it is still revealed by the DNS queries used to reach the server.  To achieve the security goals from {{Section 10.1 of !ECH}}, clients for this specification SHOULD make queries over an authenticated and encrypted connection to a DNS resolver that they trust to preserve the confidentiality of their queries and return authentic answers.
```

**Review by MikeBishop — APPROVED** [review]

File: draft-ietf-tls-svcb-ech.md (hunk 1 above)
Comment (paulwouters) [inline draft-ietf-tls-svcb-ech.md]: I would say "from disclosure in the TLS protocol by ECH"

Comment (paulwouters) [reply draft-ietf-tls-svcb-ech.md]: and amaybe "to resolve the server" instead of "to reach" which is kinda vague and weird

Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: OK, I've incorporated those changes.

Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: OK, I've incorporated a version of this wording.

File: draft-ietf-tls-svcb-ech.md [hunk 2]
```diff
@@ -96,7 +96,7 @@ A SVCB RRSet containing some RRs with "ech" and some without is vulnerable to a
...
 
-ECH ensures that TLS does not disclose the SNI, but the same information is also revealed in the DNS queries used to resolve the server's hostname.  To achieve the security goals from {{Section 10.1 of !ECH}}, clients for this specification SHOULD use an authenticated and encrypted connection to a DNS resolver that they trust to preserve the confidentiality of their queries and return authentic answers.
+ECH ensures that TLS does not disclose the SNI, but the same information is also revealed in the DNS queries used to resolve the server's hostname.  This specification does not conceal the server name from the DNS resolver.  If DNS messages are sent between the client and resolver without authenticated encryption, an attacker on this path can also learn the destination server name.
```
Comment (richsalz) [inline draft-ietf-tls-svcb-ech.md]: s/is also/can also be/ ?

Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: It sounds like you're reading "reveal" as "reveal to an attacker", whereas I meant "reveal to whoever can read it" (and at least the resolver must be able to read it!).  Maybe there's another way to clarify that...

Comment (richsalz) [reply draft-ietf-tls-svcb-ech.md]: Yes.  How about s/is also revealed/is also given/ ?

Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: I changed "is also revealed" to "is also present".

Comment (richsalz) [reply draft-ietf-tls-svcb-ech.md]: Good catch.

**Review by richsalz — APPROVED** [review]

File: draft-ietf-tls-svcb-ech.md [hunk 3]
```diff
@@ -90,12 +90,14 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
...
 
 In an idealized deployment, ECH protects the SNI with an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server. Accordingly, an attacker who can enumerate this set can always guess the encrypted SNI with probability 1/K, where K is the number of domains in the set. In practice, this probability may be increased via traffic analysis, popularity weighting, and other mechanisms.
 
+ECH ensures that TLS does not disclose the SNI, but the same information is also present in the DNS queries used to resolve the server's hostname.  This specification does not conceal the server name from the DNS resolver.  If DNS messages are sent between the client and resolver without authenticated encryption, an attacker on this path can also learn the destination server name.
```
Comment (enygren) [inline draft-ietf-tls-svcb-ech.md]:
```suggestion
ECH ensures that TLS does not disclose the SNI, but the same information is also present in the DNS queries used to resolve the server's hostname.  This specification does not conceal the server name from the DNS resolver.  If DNS messages are sent between the client and resolver without authenticated encryption, an attacker on this path can also learn the destination server name.  A similar exposure applies when the DNS resolver is closely correlated with the requesting client, as an on path attacker may be able to infer the destination server name or information about it by observing traffic to DNS authorities.
```

Comment (enygren) [reply draft-ietf-tls-svcb-ech.md]: Would we want to go into more detail, or just reference https://datatracker.ietf.org/doc/html/rfc9076 (DNS privacy considerations)?

Comment (bemasc) [reply draft-ietf-tls-svcb-ech.md]: OK, I added "A similar attack applies if the client can be linked to a request from the resolver to a DNS authority."

**Review by martinthomson — APPROVED** [review]

**Review by paulwouters — APPROVED** [review]

**Pull request merged**